Add document for sticky consumer (#5745)
apache/pulsar: site2/docs/client-libraries-java.md

("key-4", "message-4-2")
```

> By default, consumer in `Key_Shared` subscription will be assigned a fixed hash range of key automatically. If you want to specify the hash ranges of a consumer, you can using the key shared policy:

Reviewer: ```suggestion
By default, a consumer in the key_shared subscription is assigned a fixed hash range of keys automatically. If you want to specify the hash range of a consumer, you can use the key shared policy:
```

Zero conditional sentences express general truths—situations in which one thing always causes another. When you use a zero conditional sentence, you’re talking about a general truth rather than a specific instance of something. 

Example

If you don’t brush your teeth, you get cavities.
When people smoke cigarettes, their health suffers.

Author: We'd better keep `Key_Shared`, since other places also use `Key_Shared` and we have already publish the `Key_Shared` subscription in 2.4.0 release. 
A consumer can specify with multiple key hash ranges, is it ok to use `key hash ranges` instead `hash range of keys`

Reviewer: 1. For key_shared subscription, there are many inconsistencies (issues of capitalization and backtick) in Pulsar doc

* w/o backtick
![image](https://user-images.githubusercontent.com/50226895/69615827-86131300-1070-11ea-82cc-336a86adaf9b.png)

* w/ backtick
![image](https://user-images.githubusercontent.com/50226895/69615747-61b73680-1070-11ea-9e5f-098ea6bf081e.png)

**Shall we use `Key_Shared` subscription overall?**

2. Do you mean changing it to the following?

`By default, a consumer in the key_shared subscription is assigned a fixed hash range of keys automatically. If you want to specify the hash ranges of a consumer, you can use the key shared policy:`

Author: > Shall we use Key_Shared subscription overall?
Yes, it's better to use `Key_Shared`,  in the java source code file also use `Key_Shared`

> Do you mean changing it to the following?

By default, a consumer in the key_shared subscription is assigned a fixed key hash range automatically. If you want to specify the key hash ranges of a consumer, you can use the key shared policy:

Reviewer: OK, thanks for your explanations.

```java
Consumer consumer = client.newConsumer()
        .topic("my-topic")
        .subscriptionName("my-subscription")
        .subscriptionType(SubscriptionType.Key_Shared)
  			.keySharedPolicy(KeySharedPolicy.sticky().ranges(Range.of(0, 10)))
        .subscribe()
```

> The consumer with specific key hash ranges also can be called a sticky consumer. Available key hash range is [0, 65535] , consumers of the subscription must cover the whole key hash range and no overlap .

Reviewer: ```suggestion
The consumer with a specific key hash range can be called a **sticky consumer**. The available key hash range is [0, 65535]. Consumers of a subscription must cover the whole key hash range and can not overlap.
```
